### PR TITLE
Added one endtoend testcase that does the status check

### PR DESCRIPTION
**Pull request by arindamnayak**
This will create a grpc server which contains health check impl. Our testcase will use the generated binary and using actual process, it will execute it with addr pointing to grpc server and it will compare the output with expected status.

So far, I have added only one testcase, which checks the `SERVING` status.

### CODE REVIEW

File: test/healthserver/proto/health.proto [hunk 1]
```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
```
Comment (ahmetb) [inline test/healthserver/proto/health.proto:1]: you don't need this. health package comes in go.

File: test/grpc_health_probe_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package test
```
Comment (ahmetb) [inline test/grpc_health_probe_test.go:1]: no need a separate program for this. you can start a grpc server in a unit test case, and shut it down later. I just omitted doing so.

Comment (ahmetb) [reply test/grpc_health_probe_test.go:1]: the only thing we would need to do would be moving main(){} into a testable method :)

Comment (arindamnayak) [reply test/grpc_health_probe_test.go:1]: @ahmetb , thanks for the input. I thought that, we are trying to test it from outside(kind of functional) and we don't want to change the original methods. Anyway, I will fix my PR.

Comment (ahmetb) [reply test/grpc_health_probe_test.go:1]: 🤔 yeah for that we need probably to start an in-memory server with various configurations and write proper integration test suite. We can start with 1 test as you said here.

Comment (arindamnayak) [reply test/grpc_health_probe_test.go:1]: @ahmetb , I have updated code to use in-memory server and removed the custom .proto and other files. We can start with one test, I am planning to add more test. I have also included the test_util.go from the testing branch.

**Comment by ahmetb** [comment]
After some thinking, I am thinking extracting the code into a shape that's nicely testable will most likely increase the source code 3-4 times excluding the test code.

So for now I'll close this to keep things simple and easily readable. Thanks for your contribution, however.

**Pull request closed without merging**